src/lib/onlyText.ts

```diff
@@ -22,9 +21,14 @@ const onlyText = (children: ReactNode | ReactNode[]): string => {
   return Children.toArray(children).reduce((text: string, child: ReactNode): string => {
     let newText = '';
 
-    if (isValidElement(child) && hasChildren(child)) {
```

Comment:
After a bit of though I think this should only be `hasChildren(child)` since isValidElement has already been called

Reply:
No, we can't.

Okay, you could change code to this:

```ts
const onlyText = (children: ReactNode | ReactNode[]): string => {
  if (!(children instanceof Array) && !isValidElement(children)) {
    return childToString(children);
  }

  return Children.toArray(children).reduce((text: string, child: ReactNode): string => {
    let newText = '';

    const hasChildrenResult = hasChildren(child);

    if (hasChildrenResult) {
      newText = onlyText(child.props.children);
    } else if (!hasChildrenResult) {
      newText = '';
    } else {
      newText = childToString(child);
    }

    return text.concat(newText);
  }, '');
};
```

and run tests. It will fall.

Reply:
If the only changes on this pr are these, it passes the tests and coverage and will improve the amount of calls
```
    if (hasChildren(child)) {
      newText = onlyText(child.props.children);
    } else if (isValidElement(child)) {
      newText = '';
    } else {
      newText = childToString(child);
    }
```